src/App.tsx

```diff
@@ -21,75 +38,43 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            value={itemsPerPage}
+            onChange={handleItemsAmountPerPage}
           >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            {options.map((option) => {
+              return (
+                <option
+                  value={option}
+                  key={option}
+                  selected={option === itemsPerPage}
```

Comment:
Remove 'selected' attribute from the option tag as it is already controlled by value attribute of the select tag.